File: src/main/kotlin/me/kyleseven/consolereader/config/ConfigLoader.java
package me.kyleseven.consolereader.config;

import me.kyleseven.consolereader.ConsoleReader;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.configuration.file.YamlConfiguration;

import java.io.File;

public abstract class ConfigLoader {
    protected ConsoleReader plugin = ConsoleReader.getPlugin();
    protected String fileName;
    protected File configFile;
    protected FileConfiguration config;

    public ConfigLoader(String relativePath, String fileName) {
        this.fileName = fileName;
        this.configFile = new File(plugin.getDataFolder(), relativePath + File.separator + fileName);
        loadFile();
    }

    public ConfigLoader(String fileName) {
        this.fileName = fileName;
        this.configFile = new File(plugin.getDataFolder(), fileName);
        loadFile();
    }

    protected void loadFile() {
        if (!configFile.exists()) {
            plugin.getLogger().info("Creating " + fileName + " file...");
            plugin.saveResource(fileName, false);
        } else {
            plugin.getLogger().info("Loading " + fileName + " file...");
        }

        config = YamlConfiguration.loadConfiguration(configFile);
    }
}

File: src/main/kotlin/me/kyleseven/consolereader/config/ConfigLoader.kt
package me.kyleseven.consolereader.config

import me.kyleseven.consolereader.ConsoleReader
import org.bukkit.configuration.file.YamlConfiguration
import java.io.File

abstract class ConfigLoader(private val fileName: String) {
    private val plugin = ConsoleReader.instance
    private var configFile = File(plugin.dataFolder, fileName)
    protected var config = loadConfig()

    protected fun loadConfig(): YamlConfiguration {
        if (!configFile.exists()) {
            plugin.logger.info("Creating $fileName file...")
            plugin.saveResource(fileName, false)
        } else {
            plugin.logger.info("Loading $fileName file...")
        }
        return YamlConfiguration.loadConfiguration(configFile)
    }
}